File: fitboyz-frontend/src/services/ProgressService.js
import axios from "axios"

const PROGRESS_API_BASE_URL = "/api/progress"

const USER_PARAM = "?userId="
const EXERCISE_PARAM = "&name="

class ProgressService {
  getProgress(name, userId) {
    return axios.get(
      PROGRESS_API_BASE_URL + USER_PARAM + userId + EXERCISE_PARAM + name
    )
  }
  getAllProgress(userId) {
    return axios.get(PROGRESS_API_BASE_URL + "/all" + USER_PARAM + userId)
  }
  createProgress(userId, progress) {
    return axios.post(PROGRESS_API_BASE_URL + USER_PARAM + userId, progress)
  }
}

export default new ProgressService()

File: fitboyz-backend/src/main/java/com/mvd/fitboyzbackend/web/service/ProgressService.java
package com.mvd.fitboyzbackend.web.service;

import com.mvd.fitboyzbackend.db.client.*;
import com.mvd.fitboyzbackend.db.model.*;
import com.mvd.fitboyzbackend.web.dto.*;
import com.mvd.fitboyzbackend.web.manipulator.*;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ProgressService {

    private final ProgressDbClient progressDbClient;
    private final ProgressManipulator progressManipulator;
    private final UserDbClient userDbClient;
    private final CompletedExerciseDbClient completedExerciseDbClient;

    public ProgressDto createProgress(ProgressDto progressDto, Long userId) {
        Progress progress = progressManipulator.convertDtoToProgress(progressDto);
        User user = userDbClient.getUserById(userId);
        progress.setUser(user);

        progress = progressDbClient.saveRecord(progress);
        return progressManipulator.convertProgressToDto(progress);
    }

    public List<ProgressDto> getProgress(Long userId) {
        List<Progress> progresses = progressDbClient.getProgresses(userId);
        return progressManipulator.convertProgressesToDtos(progresses);
    }

    public List<CompletedExerciseDto> getExerciseProgress(String name, Long userId) {
        List<CompletedExercise> completedExercises = completedExerciseDbClient.getCompletedExercises(name, userId);
        return progressManipulator.convertCompletedExercisesToDtos(completedExercises);
    }
}
